linkcheck/checker/urlbase.py

```diff
@@ -383,15 +383,16 @@ def build_url_parts (self):
         Also checks for obfuscated IP addresses.
         """
         # check userinfo@host:port syntax
-        self.userinfo, host = urllib.parse.splituser(self.urlparts[1])
-        port = urlutil.default_ports.get(self.scheme, 0)
-        host, port = urlutil.splitport(host, port=port)
+        o = urllib.parse.urlparse("//%s" % self.urlparts[1])
+        self.userinfo = ":".join([x for x in (o.username, o.password) if x is not None])
+        port = o.port
+        if not port:
+            port = urlutil.default_ports.get(self.scheme, 0)
         if port is None:
```

Comment:
Can it ever be `None`, given the above check?

TBH I don't see how it could be None given the old code as well.

Reply:
I can't see how.